code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+#          out_of_area_case_indicator=Outofareacaseindicator,
+#          admission_date=OriginalAdmissionDate,
+#          sex_code=Gender,
+#          discharge_date=DateDischarge,
+#          discharge_to_code=DischargeReason,
+#          census_flag=CENSUSFLAG,
+#          delay_description=DELAY_DESCRIPTION)
+
+#Create age_grpgGrouping
+
+datafile<-datafile%>% mutate(age_grp=
+                               if_else(AGEATRDD<75, "18-74",
+                                       if_else(AGEATRDD>=75, "75+", " ")))
+
+
+
+
+
+#save out file for main bed days file
+ 
+
+#write_sav(datafile,paste0(filepath,"main bed days file.sav")) # save out file
+
+#set function
+#'%!in%' <- function(x,y)!('%in%'(x,y))
+
+#Save census data file
+
+
+datafile2 <- filter(datafile, !REASONFORDELAYSECONDARY %in%c("26X","46X") & CENSUSFLAG=="Y")
+
+
+#write_sav(datafile2,paste0(filepath,"main census file.sav")) # save out file
+
+
+
+# 5. Create Reason 2 Groupings ----
+
+##Sub grouping reason_grp for code 9s are currently being classed as H&SC or Pat/Fam reasons. Ensure they are just code 9s.
+##Also bring in Transport as a separate category.
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Adults with Incapacity Act", "Adults with Incapacity Act",reas2))
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION!="Adults with Incapacity Act" & reas1=="Code 9", "Other code 9 reasons(not AWI)",reas2))
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Awaiting availability of transport","Transport",reas2))
+
+
+#write_sav(datafile3,paste0(filepath,"main census file Tabs 1 3 6.sav")) # save out file
+
+### 6. Get reason / age_grp breakdown for Scotland ----
+
+datafile4<-datafile3
+datafile4$level<-"1"
+
+datafile4$areaname<-"Scotland"
+
+Scotlandbymainreasongrouping <- datafile4 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas1) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
```

Comment:
See "Style Guide" and "Incorrect Variable Names" sections in general feedback comments.

Have a look at the `summarise_at()` function. This will make the code more concise. I'll make a suggestion for this but the same can be done throughout the script where a similar pattern occurs.
```suggestion
Scotlandbymainreasongrouping <- datafile %>%
    group_by(year, MONTHFLAG, level, areaname, age_grp, reas1) %>%
    summarise_at(vars(num_pats, delay_1_to_3_days:delay_over_6_weeks, acute:notgpled),
               ~ sum(., na.rm = TRUE)) %>%
    ungroup()
```

Reply:
Agree with this idea - however `summarise_at` has been superseded by `across` in this new version of dplyr so I'd suggest using this instead.